File: src/main/java/xyz/lncvrt/galaxyboxpvp/events/BlockBreakListener.java
package xyz.lncvrt.galaxyboxpvp.events;

import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvP;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;

public class BlockBreakListener implements Listener {
    private final GalaxyBoxPvP plugin;

    public BlockBreakListener(GalaxyBoxPvP plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onBlockBreak(BlockBreakEvent event) {
        Player player = event.getPlayer();
        plugin.convertInventoryItemsPrep(player, player.getUniqueId());
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/BlockBreakListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvPPlugin

class BlockBreakListener(private val plugin: GalaxyBoxPvPPlugin) : Listener {
    @EventHandler
    fun onBlockBreak(event: BlockBreakEvent) {
        val player = event.player
        plugin.convertInventoryItemsPrep(player, player.uniqueId)
        if (event.getBlock().location.y == 100.0 && event.getBlock().world.name == "world") {
            event.isCancelled = true
        }
    }
}